.github/workflows/mixed_mode_test.yml

```diff
@@ -0,0 +1,44 @@
+name: Mixed Mode Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to test from'
+        required: true
+
+jobs:
+  gradle:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4.2.2
+        with:
+          ref: ${{ inputs.tag }}
+          ssh-key: ${{ secrets.DEPLOY_KEY }}
+      - name: Setup Base Environment
+        uses: ./actions/setup-base-env
+      - name: Setup FDB
+        uses: ./actions/setup-fdb
+
+      # Push a version bump back to main. There are failure scenarios that can result
+      # in published artifacts but an erroneous build, so it's safer to bump the version
+      # at the beginning
+      - name: Configure git
+        run: |
+          git config --global user.name 'FoundationDB CI'
+          git config --global user.email 'foundationdb_ci@apple.com'
+      - name: Run Gradle Test
+        uses: ./actions/gradle-test
+        with:
+          gradle_command: mixedModeTest
+          gradle_args: -PreleaseBuild=false -PpublishBuild=false
+      - name: Checkout Main
+        run: git checkout main
```

Comment:
Did you get a chance to test this? I believe when I tried to do something similar with a test repo, the equivalent step here failed with a "reference not found", and I needed to: (1) checkout `main` as the first step, (2) call `checkout` again to switch to the actual target, and (3) checkout `origin/main` instead of `main`

Reply:
I ran it through the test repo, and got a passing build that updated the release notes as expected.